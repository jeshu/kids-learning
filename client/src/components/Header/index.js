import React, { useState } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { AppBar, Toolbar, IconButton, Typography, Container } from '@material-ui/core';
import SettingPopup from '../SettingsPopup';
import { Settings } from '@material-ui/icons';
const useStyles = makeStyles(theme => ({
    root: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        flexGrow: 1,
    },
}))

function Header({ onSettingsUpdate }) {
    const [settingStatus, setSettingStatus] = useState(false)
    const classes = useStyles();
    const showSettings = () => {
        setSettingStatus(true);
    };

    const hideSettings = () => {
        setSettingStatus(false);
    };
    return (
        <div data-testid="header-component">
            <AppBar position="static">
                <Container>

                    <Toolbar>
                        <Typography variant="h6" className={classes.title} data-testid="logo">Kids Learning</Typography>
                        <IconButton color="inherit" onClick={showSettings} data-testid="settings-btn">
                            <Settings />
                        </IconButton>
                    </Toolbar>
                </Container>
            </AppBar>
            <SettingPopup show={settingStatus} onClose={hideSettings} onSettingsUpdate={onSettingsUpdate} />
        </div>
    )
}

export default Header
