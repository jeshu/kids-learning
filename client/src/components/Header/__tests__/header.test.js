import React from 'react'
import Header from '../index';
import {render, fireEvent, screen, waitForDomChange} from '@testing-library/react';

describe('Header component test', () => {
    it('should render the header component wihtout errors', () => {
        const {asFragment} = render(<Header />);
        expect(asFragment()).toMatchSnapshot();
        expect(screen.getByTestId("header-component")).toBeInTheDocument();
    });

    it('should should render a logo in header', ()=>{
        render(<Header />)
        expect(screen.getByTestId("logo")).toBeInTheDocument();
    });

    it('should have settings button', ()=>{
        render(<Header />)
        expect(screen.getByTestId("settings-btn")).toBeInTheDocument();
    })

    it('should click on setting button open the settingPopup', async ()=> {
        const setState = jest.fn();
        const useStateSpy = jest.spyOn(React, "useState");
        useStateSpy.mockImplementation((init) => [init, setState]);
        render(<Header />)
        fireEvent.click(screen.getByTestId("settings-btn"));
        expect(setState).toHaveBeenCalledWith(true);
        
    });
})
