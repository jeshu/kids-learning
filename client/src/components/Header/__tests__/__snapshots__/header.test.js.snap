// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test should render the header component wihtout errors 1`] = `
<DocumentFragment>
  <div
    data-testid="header-component"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
          data-testid="logo"
        >
          Kids Learning
        </h6>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="settings-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h20v20H0V0z"
                fill="none"
                transform="scale(1.2, 1.2)"
              />
              <path
                d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                transform="scale(1.2, 1.2)"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
